parsec/mca/device/device.c

```diff
@@ -147,9 +156,10 @@ int parsec_get_best_device( parsec_task_t* this_task, double ratio )
          * other options to have some load balance */
         if( -1 != prefer_index ) {
             best_index = prefer_index;
-            /* we still prefer this device, until it is twice as loaded as the
+            /* we still prefer this device, until it is load_balance_skew as loaded as the
              * real best load balance device */
-            best_weight = load_balance_skew * (parsec_device_load[prefer_index] + ratio * parsec_device_sweight[prefer_index]);
+            dev = parsec_mca_device_get(best_index);
```

Comment:
Same thing here, the skew should be on the resulting load balancing (that is, `skew*(device->load+time_estimate(task,device))`).